### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by brytebee**
### Processing data 👤
In this PR, I have achieved the following:
- Updated the models.
- Used models to insert data
- Created custom method.
   - A method that returns the 3 most recent posts for a given user.
   - A method that updates the posts counter for a user.
   - A method which returns the 5 most recent comments for a given post.
   - A method that updates the comments counter for a post.
   - A method that updates the likes counter for a post.
   - A method that returns current_user in the ApplicationController.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
 class ApplicationController < ActionController::Base
+  def current_user
+    user.first
```
Comment (Lameck1) [inline app/controllers/application_controller.rb]: - [ ] Well done for implementing this. However, to refer to the User class  we need to rewrite `user` starting with a capital letter. This is just a Rails convention and it is how it works. Do something like this:
  
  ```ruby
  User.first
  ```

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,4 +1,12 @@
 class Comment < ApplicationRecord
+  after_save :update_comment_counter
+
   belongs_to :author, class_name: 'User', foreign_key: 'author_id'
```
Comment (Lameck1) [inline app/models/comment.rb]: - [OPTIONAL] I wish we could realize that there is no need of defining the foreign key here. The User model has taken care of this already. Now, this becomes redundant. Having told rails that this model belongs to the `author`, the primary key will automatically be created.
```suggestion
  belongs_to :author, class_name: 'User'
```

File: app/models/like.rb [hunk 3]
```diff
@@ -1,4 +1,10 @@
 class Like < ApplicationRecord
+  after_save :update_likes_counter
+
   belongs_to :author, class_name: 'User', foreign_key: 'author_id'
```
Comment (Lameck1) [inline app/models/like.rb]: - [OPTIONAL] Same issue as above, please consider.

File: app/models/post.rb [hunk 4]
```diff
@@ -1,5 +1,17 @@
 class Post < ApplicationRecord
+  after_save :increment_posts_counter
+
   belongs_to :author, class_name: 'User', foreign_key: 'author_id'
```
Comment (Lameck1) [inline app/models/post.rb]: - [OPTIONAL] Same issue as above, please consider.

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi @brytebee, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

# Highlights
- Usage of Rails Active Callbacks ✔️ 
- Professional README ✔️
- Clean code ✔️ 
- Good commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (brytebee) [reply app/controllers/application_controller.rb]: Done!

File: app/models/comment.rb (hunk 2 above)
Comment (brytebee) [reply app/models/comment.rb]: Done!

File: app/models/like.rb (hunk 3 above)
Comment (brytebee) [reply app/models/like.rb]: Done!

File: app/models/post.rb (hunk 4 above)
Comment (brytebee) [reply app/models/post.rb]: Done. Thanks, @Lameck1 for the insight!

**Review by Lameck1 — APPROVED** [review]
Hi, @brytebee,

Tasks for this project have been successfully implemented 💯.

### Highlights
- Usage of Rails Active Callbacks ✔️
- Professional README ✔️
- Clean code ✔️ 
- Good commit messages ✔️ 

All the best with the next project.

You are free to merge this branch :shipit:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**